docs/pages/agents/build-an-agent.mdx

```diff
@@ -186,6 +186,76 @@ Here are some security best practices:
+
+For example, consider deploying your agent to these platforms:
+- Local development: Creates an installation
+- Railway: Creates another installation
+- Production server: Creates another installation
+
+Your agent can have up to 5 active installations before you need to revoke one to add another. Installations only accumulate for agent deployments using the same XMTP network environment, such as `local`, `dev`, or `production`.
+
+For example, if you deploy your agent across these network environments, you will have 3 inboxes, each with 1 installation:
+- Local development: `local` network
+- Railway: `dev` network  
+- Production server: `production` network
+
+<div>
+  <img src="https://raw.githubusercontent.com/xmtp/docs-xmtp-org/refs/heads/main/docs/pages/img/agent-install-sep.png" width="500px" />
+</div>
+
+If you deploy your agent to this same network environment, you have 1 inbox with 3 installations:
+- Local development: `production` network
+- Railway: `production` network
+- Production server: `production` network
+
+<div>
+  <img src="https://raw.githubusercontent.com/xmtp/docs-xmtp-org/refs/heads/main/docs/pages/img/agent-install.png" width="500px" />
+</div>
+
+Here are some best practices for agent installation management:
+
+- Configure your deployments to preserve the agent's database
+- Consider how many platforms you'll need before hitting the 5-installation limit
+- Keep track of which installation corresponds to which platform
+- Regularly [check how many installations your agent has](#test-and-debug-an-agent)
+- [Revoke installations](#revoke-agent-installations) from platforms you no longer use
+
+### Common agent installation scenarios
+
+#### Deploy to a new platform
+
+When you deploy your agent to a new platform (e.g., from Railway to Fly.io):
+- Your agent creates a new installation for the new platform
+- If you already have 5 installations, you'll need to [revoke one first](#revoke-agent-installations)
+- The new installation will start fresh without conversation history
+
+#### Update an existing deployment
+
+When you redeploy your agent to the same platform:
+- If the platform preserves the database, the same installation continues working
+- If the platform doesn't preserve the database, a new installation is created
+
+### Revoke agent installations
+
+:::warning[Important]
+
+Revoking an installation is permanent. You cannot recover access to a revoked installation.
+
+:::
+
+When you revoke an agent installation, it can no longer send or receive messages. However, you can still access the local database, if its not encrypted. You can also still access the message history up to the point that the installation was revoked.
```

Comment:
```suggestion
When you revoke an agent installation, it can no longer send or receive messages. However, you can still access the local database if it's not encrypted. You can also still access the message history up to the point that the installation was revoked.
```